cmd/k8sm-executor/main.go

```diff
@@ -54,135 +48,120 @@ var (
-		endpoint = os.Getenv("DOCKER_HOST")
-	} else {
-		endpoint = "unix:///var/run/docker.sock"
-	}
-	log.Infof("Connecting to docker on %s", endpoint)
-
-	return endpoint
-}
-
-func getHostname() string {
-	hostname := []byte(*hostnameOverride)
-	if string(hostname) == "" {
-		// Note: We use exec here instead of os.Hostname() because we
-		// want the FQDN, and this is the easiest way to get it.
-		fqdn, err := exec.Command("hostname", "-f").Output()
-		if err != nil {
-			log.Fatalf("Couldn't determine hostname: %v", err)
-		}
-		hostname = fqdn
-	}
-	return strings.TrimSpace(string(hostname))
-}
-
-func getApiserverClient() (*client.Client, error) {
-	clientConfig := client.Config{}
-	if *authPath != "" {
-		authInfo, err := clientauth.LoadFromFile(*authPath)
-		if err != nil {
-			return nil, err
-		}
-		clientConfig, err = authInfo.MergeWithConfig(clientConfig)
-		if err != nil {
-			return nil, err
-		}
-	}
-	// TODO: adapt Kube client to support LB over several servers
-	if len(apiServerList) > 1 {
-		log.Infof("Mulitple api servers specified.  Picking first one")
-	}
-	clientConfig.Host = apiServerList[0]
-	if c, err := client.New(&clientConfig); err != nil {
-		return nil, err
-	} else {
-		return c, nil
-	}
+	flag.Var(&clusterDNS, "cluster_dns", "IP address for a cluster DNS server.  If set, kubelet will configure all containers to use this for DNS resolution in addition to the host's DNS servers")
 }
 
 func main() {
 
 	flag.Parse()
-	util.InitLogs() // TODO(jdef) figure out where this actually sends logs by default
+	util.InitLogs()
 	defer util.FlushLogs()
 	rand.Seed(time.Now().UTC().UnixNano())
 
 	verflag.PrintAndExitIfRequested()
 
-	etcd.SetLogger(util.NewLogger("etcd "))
-
-	// Make an API client if possible.
-	if len(apiServerList) < 1 {
-		log.Info("No api servers specified.")
-	} else {
-		if apiClient, err := getApiserverClient(); err != nil {
-			log.Errorf("Unable to make apiserver client: %v", err)
-		} else {
-			// Send events to APIserver if there is a client.
-			log.Infof("Sending events to APIserver.")
-			record.StartRecording(apiClient.Events(""), "kubelet")
-		}
+	if err := util.ApplyOomScoreAdj(*oomScoreAdj); err != nil {
+		log.Info(err)
+	}
+
+	kcfg := standalone.KubeletConfig{
+		Address:                 address,
+		AuthPath:                *authPath,
+		ApiServerList:           apiServerList,
+		AllowPrivileged:         *allowPrivileged,
+		HostnameOverride:        *hostnameOverride,
+		RootDirectory:           *rootDirectory,
+		NetworkContainerImage:   *networkContainerImage,
+		SyncFrequency:           *syncFrequency,
+		RegistryPullQPS:         *registryPullQPS,
+		RegistryBurst:           *registryBurst,
+		MinimumGCAge:            *minimumGCAge,
+		MaxContainerCount:       *maxContainerCount,
+		ClusterDomain:           *clusterDomain,
+		ClusterDNS:              clusterDNS,
+		Port:                    *port, // TODO(nnielsen): Don't hardwire port, but use port from resource offer.
```

Comment:
the point was that the port should come from the mesos `ports` resource range, and not from the command line. i'll update the wording since it's odd now that the code has shifted.